applications/crossbar/src/modules/cb_fmc_devices.erl

```diff
@@ -0,0 +1,285 @@
+            case wh_doc:is_soft_deleted(Doc) of
+                'true' ->
+                    cb_context:add_validation_error(<<"fmc">>, <<"not_found">>, wh_json:from_list([{<<"message">>, <<"Device with this ID wasn't found">>}]), Context);
+                'false' ->
+                    AccountId = cb_context:account_id(Context),
+                    case wh_json:get_value(<<"account_id">>, Doc) of
+                        AccountId ->
+                            DeviceId = wh_json:get_value(<<"device_id">>, Doc),
+                            {'ok', DeviceDoc} = couch_mgr:open_doc(cb_context:account_db(Context), DeviceId),
+                            case wh_json:get_value(<<"device_type">>, DeviceDoc) of
+                                <<"cellphone">> ->
+                                    Number = wh_json:get_value([<<"call_forward">>, <<"number">>], DeviceDoc),
+                                    case is_existing_fmc(Number, FMCValue, Id) of
+                                        'true' ->
+                                            cb_context:add_validation_error(<<"fmc">>, <<"unique">>, wh_json:from_list([{<<"message">>, <<"FMC Device with this A-Number and FMC Value already exists">>}]), Context);
+                                        'false' ->
+                                            PrivJObj = wh_json:private_fields(Doc),
+                                            NewDoc = wh_json:merge_jobjs(PrivJObj, cb_context:doc(Context)),
+                                            cb_context:set_doc(Context, NewDoc)
+                                    end;
+                                _ ->
+                                    cb_context:add_validation_error(<<"fmc">>, <<"invalid">>, wh_json:from_list([{<<"message">>, <<"Invalid device type">>}]), Context)
+                            end;
+                        _ ->
+                            cb_context:add_validation_error(<<"fmc">>, <<"not_found">>, wh_json:from_list([{<<"message">>, <<"FMC Device with this ID wasn't found">>}]), Context)
+                    end
+            end;
+        _ ->
+            cb_context:add_validation_error(<<"fmc">>, <<"not_found">>, wh_json:from_list([{<<"message">>, <<"FMC Device with this ID wasn't found">>}]), Context)
+    end.
+
+is_existing_fmc(Number, FMCValue, 'undefined') ->
+    {'ok', JObjs} = couch_mgr:get_all_results(?WH_FMC_DB, ?CB_LIST),
+    Doc = [wh_json:get_value(<<"value">>, JObj) || JObj <- JObjs],
+    Doc1 = [{wh_json:get_value(<<"a_number">>, JObj), wh_json:get_value(<<"x_fmc_value">>, JObj)} || JObj <- Doc],
+    lists:member({Number, FMCValue}, Doc1);
+is_existing_fmc(Number, FMCValue, FMCDocId) ->
+    {'ok', JObjs} = couch_mgr:get_all_results(?WH_FMC_DB, ?CB_LIST),
+    Doc = [wh_json:get_value(<<"value">>, JObj) || JObj <- JObjs],
+    Doc1 = [{wh_json:get_value(<<"a_number">>, JObj), wh_json:get_value(<<"x_fmc_value">>, JObj)} || JObj <- Doc, wh_json:get_value(<<"id">>, JObj) =/= FMCDocId],
```

Comment:
Try not to duplicate code at a minimum please.

Also, get doc id with `wh_doc:id/1`